src/go/rpk/pkg/publicapi/publicapi.go

```diff
@@ -33,6 +33,15 @@ func newAuthInterceptor(token string) connect.UnaryInterceptorFunc {
 	}
 }
 
+func newReloadingAuthInterceptor(f func() string) connect.UnaryInterceptorFunc {
```

Comment:
is a specialized interceptor, that does not use a static token. instead, it can call a function that provides the token (which can be refreshed in the background)